### DISCUSSION TITLE
Why can't I set style to my usercontrol?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#15012] [Q&A] [answered]
I created a usercontrol and set its default BorderBrush to Blue like this:
```
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
			 xmlns:Local="using:AvaloniaApplication1"
             x:Class="AvaloniaApplication1.UserControl1">
	<UserControl.Styles>
		<Style Selector="Local|UserControl1">
			<Setter Property="BorderBrush" Value="Blue"></Setter>
		</Style>
	</UserControl.Styles>
	<Border BorderBrush="{Binding $parent[UserControl].BorderBrush}" BorderThickness="2">
		
	</Border>
</UserControl>
```

And then I use it in the Window and in some window the BorderBrush of this usercontrol should set to red. I do it like this:
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication1.MainWindow"
		xmlns:Local="using:AvaloniaApplication1"
        Title="AvaloniaApplication1">
	<Window.Styles>
		<Style Selector="Local|UserControl1">
			<Setter Property="BorderBrush" Value="Red"></Setter>
		</Style>
	</Window.Styles>	
	<Local:UserControl1 Width="100" Height="100">
    </Local:UserControl1>
</Window>
```

![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/17151f5c-4b3c-482f-a31b-2bccf724c8c4)

Whereas, it still is the default BorderBrush color.

I guess there is a problem with setting the default BorderBrush value, which causes the Style in Window to fail to rewrite the default BorderBrush value. But I'm not sure if that's the problem, and I don't know of any other way for me to set the default BorderBrush value.

**Comment by thevortexcloud** [comment, marked as answer]
The style at the control level is a higher priority since it's more local. The default controls get around this using templated controls and template bindings. If you don't want to make a templated control (which it sounds like you possibly want to do anyway), move the default style to a higher level than the window. EG in your application styles. The window style will then be the most local style.

**Reply by kyuranger** [reply]
Thank you very much, I have a deeper understanding of the difference between templatedcontrol and usercontrol.
